src/menus/hotKeyMenu.cpp

```diff
@@ -0,0 +1,207 @@
+    next_page = new GuiButton(this, "PAGE_RIGHT", "->", [this]()
+    {
+        HotKeyMenu::pageHotkeys(-1);
+    });
+    next_page->setPosition(-50, 50, ATopRight)->setSize(50, 50)->disable();
+
+    // frame with keys
+    frame = new GuiPanel(this, "HELP_FRAME");
+    frame->setPosition(50, FRAME_TOP, ATopLeft)->setSize(600, FRAME_HEIGHT);
+
+    // TODO: This label is redundant with the category selector
+    // category_label = new GuiLabel(frame, "Category_Label", category, 30);
+    // category_label->addBackground()->setPosition(50, 10, ATopLeft)->setSize(500, 50);
+
+    // initial category listing
+    HotKeyMenu::setCategory(1);
+
+    // Bottom GUI.
+    // Back button.
+    (new GuiButton(this, "BACK", tr("options", "Back"), [this]()
+    {
+        // Close this menu, stop the music, and return to the main menu.
+        destroy();
+        soundManager->stopMusic();
+        returnToOptionMenu();
+    }))->setPosition(50, -50, ABottomLeft)->setSize(150, 50);
+
+    // Update Hotkey Values
+    (new GuiButton(this, "UPDATE", tr("options", "Save"), [this]()
+    {
+        HotKeyMenu::updateHotKeys();
+    }))->setPosition(200, -50, ABottomLeft)->setSize(150, 50);
+}
+
+void HotKeyMenu::onKey(sf::Event::KeyEvent key, int unicode)
+{
+    switch(key.code)
+    {
+        //TODO: This is more generic code and is duplicated.
+        case sf::Keyboard::Escape:
+        case sf::Keyboard::Home:
+            destroy();
+            returnToOptionMenu();
+            break;
+        default:
+            break;
+    }
+}
+
+void HotKeyMenu::setCategory(int cat)
+{
+    // remove old entries
+    for (GuiTextEntry* text : text_entries){
+        text->destroy();
+    }
+    text_entries.clear();
+    for (GuiLabel* label : label_entries){
+        label->destroy();
+    }
+    label_entries.clear();
+
+    // reset hotkey frame size and position
+    int frame_width = FRAME_INITIAL_WIDTH;
+    frame->setPosition(50, FRAME_TOP, ATopLeft)->setSize(frame_width, FRAME_HEIGHT);
+
+    // get chosen category
+    category_index = cat;
+    category = category_list[cat];
+
+    int top = FRAME_PADDING;
+    int left = FRAME_PADDING;
+
+    hotkey_list = hotkeys.listAllHotkeysByCategory(category);
+
+    // TODO: This label is redundant with the category selector
+    // category_label->setText(category);
+
+    // begin filling of Hotkey listing for category
+    for (std::pair<string,string> item : hotkey_list) {
+
+        label_entries.push_back(new GuiLabel(frame, "NAME_LABEL", item.first.append(" = "), 30));
+        label_entries.back()->setAlignment(ACenterRight)->setPosition(left, top, ATopLeft)->setSize(400, 50);
+
+        text_entries.push_back(new GuiTextEntry(frame, "HotKey_Entry", item.second));
+        text_entries.back()->setTextSize(30)->setPosition(left + 400, top, ATopLeft)->setSize(100, 50);
+        top += 50;
+
+        // This keeps adding columns even if they don't fit on the screen
+        if (top > 650) {
```

Comment:
```suggestion
        if (top > 650) 
        {
```

I'm also more of a fan of calculating things (since now if the layout ever changes, the 650 might not be correct). Although in this case I'm not so sure how easy it is.

Reply:
Me neither, and I called it out in the PR description — the layout structure was lifted from the 2017 PR and needs updating. It should be easier to use GuiAutoLayouts for organizing the three main "rows" of title/category nav, hotkeys, and back/save nav.

The challenge (for me at least) is implementing an AutoLayout with [`LayoutVerticalColumns`](https://github.com/daid/EmptyEpsilon/blob/master/src/gui/gui2_autolayout.cpp#L55) for the hotkeys, it's not one I remember using before, so I'm not 100% sure if it'll do what I think it will in this case.

Reply:
Banged my head on trying to make it an autolayout but haven't figured it out yet, I can't wrap my head around some of the behavior when an element becomes an autolayout and wind up with overlapping items, or can't select text fields or buttons because (I think) some container is on top blocking the click from passing through.

Worked around it for now by at least abstracting out the hardcoded values to reused constants set in the header.